sympy/physics/quantum/qexpr.py

```diff
@@ -98,7 +98,10 @@ class QExpr(Expr):
 
     @property
     def free_symbols(self):
-        return set([self])
+        fsyms = []
```

Comment:
These changes were needed for test_spins to pass without errors.  `canonical_variables` in `Basic` makes use of the `.name` property for all free variables.  However, QExpr creates compound 'free' variables that 
have no simple names, and thus are not canonized.  `canonical_variables` are needed for hashing
sums or integrals.  I don't know enough about the quantum files to generate a good solution to this problem.  I hope somebody can help out.